test/ably/restchannelpublish_test.py

```diff
@@ -111,10 +109,118 @@ def test_publish_various_datatypes_binary(self):
+        messages2 = existing.history().current
+        self.assertIsNotNone(messages2, msg="Expected non-None messages from existing")
+        messageContents2 = dict((m.name, m.data) for m in messages2)
+
+        self.assertEqual(messageContents1["msg"], "msgData")
+        self.assertEqual(messageContents2["msg"], "msgData")
+
+    def test_create_channel_options(self):
+        channelName = "name"
+        options = ChannelOptions(encrypted = True)
+        withOptions = TestRestChannelPublish.ably.channels.get(channelName, options=options)
+        self.assertTrue(withOptions.encrypted)
+        options = ChannelOptions(encrypted = False)
+        unencryptedWithOptions = TestRestChannelPublish.ably.channels.get(channelName, options=options)
+        self.assertFalse(withOptions.encrypted)
+        self.assertFalse(unencryptedWithOptions.encrypted)
+
+    def test_channel_exists(self):
+        channelName1 = "name"
+        channelName2 = "name2"
+        channel1 = TestRestChannelPublish.ably.channels.get(channelName1)
+        channel2 = TestRestChannelPublish.ably.channels.get(channelName2)
+        self.assertTrue(TestRestChannelPublish.ably.channels.exists(channelName1))
+        self.assertFalse(TestRestChannelPublish.ably.channels.exists("noChannelByThisName"))
+
+        names = TestRestChannelPublish.ably.channels.names()
+        channelCount = len(names)
+        self.assertEqual(channelCount, 2)
+        self.assertEqual(names[0], channelName1)
+        self.assertEqual(names[1], channelName2)
+
+    def test_publish_status(self):
+      #todo finish.
+        channel = TestRestChannelPublish.ably.channels["channel"]
+        self.assertTrue(channel.publish("msg", "data").ok)
+        #options= channel.ably.options
+        #options.host="not.a.real.host"
+        #channel.ably.http.options = options
+        #errorResponse =channel.publish("msg", "data")
+        #self.assertFalse(errorResponse.ok)
+
+    def test_release_channel(self):
+
+        channelName1 = "name"
+        channelName2 = "name2"
+        channel1 = TestRestChannelPublish.ably.channels.get(channelName1)
+        channel2 = TestRestChannelPublish.ably.channels.get(channelName2)
+        self.assertTrue(TestRestChannelPublish.ably.channels.exists(channelName1))
+        TestRestChannelPublish.ably.channels.release(channelName1)
+        names = TestRestChannelPublish.ably.channels.names()
+        channelCount = len(names)
+        self.assertEqual(channelCount, 1)
+        self.assertEqual(names[0], channelName2)
+
+    def test_publish_status(self):
+        channel1 = TestRestChannelPublish.ably.channels.get("channelName")
+        response = channel1.publish("msg", "data")
+        self.assertEqual(response.error, None)
+        restrictedAbly = AblyRest(RestSetup.testOptions(1))
+        restrictedChannel = restrictedAbly.channels.get("restrictedChannel")
+        response =restrictedChannel.publish("msg", "data")
```

Comment:
Does this mean that if you publish to a channel using REST, and it fails, an exception is not raised but instead a response object is returned?  I am not sure this is an API we want to support.  In Python when programming synchronous code, I don't think people would expect this.